lib/rack/query_parser.rb

```diff
@@ -234,20 +211,37 @@ def params_hash_has_key?(hash, key)
       true
     end
 
-    def check_query_string(qs, sep)
-      if qs
-        if qs.bytesize > @bytesize_limit
-          raise QueryLimitError, "total query size (#{qs.bytesize}) exceeds limit (#{@bytesize_limit})"
-        end
+    def each_query_pair(qs, separator, unescaper = nil)
+      return enum_for(:each_query_pair, qs, separator, unescaper) unless block_given?
```

Comment:
I know this is a pattern used by a lot of enumerable methods, but I'm not a fan. This is a private method, can we remove this? I'll include a suggestion for fixing the single affected caller.

```suggestion
```